props/prop_base.h

```diff
@@ -435,12 +432,57 @@ class PropBase : CommandParser, Looper, protected SaberBase {
 #else
     SaberBase::SetVariation(0);
 #endif
+  }
 
+  // Select preset (font/style)
+  virtual void SetPreset(int preset_num, bool announce) {
+    TRACE(PROP, "start");
+    bool on = BladeOff();
+    SaveColorChangeIfNeeded();
+    // First free all styles, then allocate new ones to avoid memory
+    // fragmentation.
+    UnsetCurrentStyles();
+    if (announce) {
+      AnnouncePreset();
+    }
+    SetBladeLength();
+    SetBladeStyle();
     if (on) On();
     if (announce) SaberBase::DoNewFont();
     TRACE(PROP, "end");
   }
 
+  // Update Blade Style (no On/Off for use in Edit Mode)
+  void UpdateStyle(int preset_num) {
+    TRACE(PROP, "start");
+    SaveColorChangeIfNeeded();
+    // First free all styles, then allocate new ones to avoid memory
+    // fragmentation.
+    UnsetCurrentStyles();
+    SetBladeLength();
+    SetBladeStyle();
+    TRACE(PROP, "end");
+  }
+
+  // Update Font / Save Style in Edit Mode, skips Preon effect (except for Preon Editing previews) using FastOn
+  void UpdateFont(int preset_num, bool preon) {
+    TRACE(PROP, "start");
+    bool on = BladeOff();
+    SaveColorChangeIfNeeded();
+    // First free all styles, then allocate new ones to avoid memory
+    // fragmentation.
+    UnsetCurrentStyles();
```

Comment:
Not sure why we're doing UnsetCurrentStyles() / SetBladeStyle() if we're only changing the font?

Reply:
To clarify further, the initial use for UpdateFont was the Edit Font function but as the IntArg and other pieces were added I used UpdateFont in other places because it was skipping the NEWFONT and PREON sounds and effects which didn't make sense during certain edits (except for Preon related edits) so rather than make more functions that did the same thing UpdateFont was used.  It was explained in the comment on the function but wanted to give more background to explain further.